plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -929,7 +929,7 @@ public void createTable(Table table)
         catch (AlreadyExistsException e) {
             throw new TableAlreadyExistsException(new SchemaTableName(table.getDbName(), table.getTableName()));
         }
-        catch (NoSuchObjectException e) {
+        catch (NoSuchObjectException | InvalidObjectException e) {
```

Comment:
I don't think throwing SchemaNotFoundException for all InvalidObjectException is a good idea. For instance, invalid table name throws InvalidObjectException.

Reply:
Maybe we should evaluate, in case of dealing with an `InvalidObjectException` whether the schema actually exists and act accordingly.

Reply:
updated accordingly